cluster/manifest/load.go

```diff
@@ -13,21 +13,47 @@ import (
 	manifestpb "github.com/obolnetwork/charon/cluster/manifestpb/v1"
 )
 
-// Load loads a cluster manifest from disk. It supports both legacy lock files and raw DAG files.
-// TODO(xenowits): Refactor, load either from manifest or lock file, see https://github.com/ObolNetwork/charon/issues/2334.
-func Load(file string, lockCallback func(cluster.Lock) error) (*manifestpb.Cluster, error) {
-	b, err := os.ReadFile(file)
+// LoadMetadata represents the result of loading a cluster from cluster manifest or legacy lock file.
+type LoadMetadata struct {
+	Filename     string // Name of the file from which the cluster was loaded
+	IsLegacyLock bool   // True if cluster was loaded from legacy lock file
+}
+
+// Load loads a cluster from disk. It supports reading from both cluster manifest and legacy lock files.
+// If both files are provided, it first reads the manifest file before reading the legacy lock file.
+// TODO(xenowits): Remove loading from legacy lock when we fully adopt mutable cluster manifests.
+func Load(manifestFile, legacyLockFile string, lockCallback func(cluster.Lock) error) (*manifestpb.Cluster, LoadMetadata, error) {
```

Comment:
Instead of returning a `LoadMetadata`, would a simple `bool` work?

Caller know at what path manifest and lock file live, so just telling them "hey I loaded a legacy lock file" should be enough IMO.

Reply:
Yeah, that would be simpler, lm update it